### DISCUSSION TITLE
[WASM] GitHub actions fails to build RELEASE

### DISCUSSION THREAD

**Discussion by DierkDroth** [unoplatform/uno#12773] [Q&A] [answered]
I'm finally getting me feet wet and am trying to build UNO WASM project in RELEASE mode using GitHub acttions. Unfortunately that fails with errors below.

I have these RELEASE settings my project file:
```
	<PropertyGroup Condition="'$(Configuration)'=='Release'">
		<DefineConstants>$(DefineConstants)</DefineConstants>
		<UnoXamlResourcesTrimming>true</UnoXamlResourcesTrimming>
		<WasmShellMonoRuntimeExecutionMode>InterpreterAndAOT</WasmShellMonoRuntimeExecutionMode>
		<WasmShellILLinkerEnabled>true</WasmShellILLinkerEnabled>
		<WasmShellWebAppBasePath>/app/wasm/Latest/</WasmShellWebAppBasePath>
	</PropertyGroup>
```

For the sake of trying I made them matching DEBUG
```
	<PropertyGroup Condition="'$(Configuration)'=='Release'">
		<DefineConstants>$(DefineConstants)</DefineConstants>
		<WasmShellILLinkerEnabled>false</WasmShellILLinkerEnabled>
		<WasmShellWebAppBasePath>/app/wasm/Latest/</WasmShellWebAppBasePath>
	</PropertyGroup>
```
... and that worked -> problem was gone.

So, I'm suspicious about a resource problem regarding the GitHub runner, may be similar to [here](https://github.com/unoplatform/uno/discussions/12669), but I'm not sure.

I could try creating a binlog, but I would not even know how to access the binlog file, since the build fails and nothing would be uploaded (I'm anything but a GitHub actions expert).

Would you have any idea what else I could try before looking into the [GitHub larger runners ](https://docs.github.com/en/actions/using-github-hosted-runners/about-larger-runners)(which come with more resources)?

Thanks in advance

```
C:\Users\runneradmin\.nuget\packages\uno.wasm.bootstrap\7.0.24\build\Uno.Wasm.Bootstrap.targets(206,3): error : System.Exception: Failed to run emscripten. More details are available in diagnostics logging mode or using the MSBuild /bl switch, see https://aka.platform.uno/msbuild-troubleshoot. [D:\a\unusual-src\unusual-src\Unusual.Wasm\Unusual.Wasm.csproj]
C:\Users\runneradmin\.nuget\packages\uno.wasm.bootstrap\7.0.24\build\Uno.Wasm.Bootstrap.targets(206,3): error :    at Uno.Wasm.Bootstrap.ShellTask_vf335ab6c9c31be2d3db727c09607aaf84b8023b1.RunPackager() in D:\a\1\s\src\Uno.Wasm.Bootstrap\ShellTask.cs:line 918 [D:\a\unusual-src\unusual-src\Unusual.Wasm\Unusual.Wasm.csproj]
C:\Users\runneradmin\.nuget\packages\uno.wasm.bootstrap\7.0.24\build\Uno.Wasm.Bootstrap.targets(206,3): error :    at Uno.Wasm.Bootstrap.ShellTask_vf335ab6c9c31be2d3db727c09607aaf84b8023b1.Execute() in D:\a\1\s\src\Uno.Wasm.Bootstrap\ShellTask.cs:line 262 [D:\a\unusual-src\unusual-src\Unusual.Wasm\Unusual.Wasm.csproj]
Error: Process completed with exit code 1.
```

**Comment by DierkDroth** [comment]
Update:
I switched my GitHub workflow from `windows-latest` to `ubuntu-latest` (cost saving + trying a different approach) and now am getting the error below (DEBUG build works fine as well on `ubuntu-latest`).

Could this be the same cause?
```
MSBUILD : error MSB4166: Child node "2" exited prematurely. Shutting down. Diagnostic information may be found in files in "/tmp/MSBuildTemprunner/" and will be named MSBuild_*.failure.txt. This location can be changed by setting the MSBUILDDEBUGPATH environment variable to a different directory.
Error: The operation was canceled.
```

**Reply by DierkDroth** [reply]
Update:
I now provisioned an Ubuntu large runner with 32GB and **that worked** (16GB didn't do it): the `Build` part finished in ~31 minutes.

So, I wonder if there are any improvements with .NET 8 in the cards? Is this something which actually is falling Microsoft court at all?!?

**Reply by jeromelaban** [reply, marked as answer]
See https://github.com/unoplatform/uno/discussions/12669#discussioncomment-6244901. This is an LLVM performance issue in the [LTO phase](https://emscripten.org/docs/optimizing/Optimizing-Code.html#lto), through emscripten when using AOT without a profile.

**Comment by DierkDroth** [comment]
Thanks for your feedback @jeromelaban. I already referenced the case link you provided above.

Anyway, appears there is not much which could be done if I didn't want to use profile guided AOT
